packages/data-grid/src/quickFilter/QuickFilter.jsx

```diff
@@ -30,11 +30,28 @@ const propTypes = {
 };
 
 function QuickFilter({ filters, onFilterSelect }) {
+  /** Used for the options array when there is no items to filter by */
+  const emptyFilterOption = [
+    {
+      label: 'No items to filter by',
+      disabled: true,
+    },
+  ];
+
   return (
     <div className="dg-quick-filter-container">
       <div className="dg-quick-filter-filters">
         {filters.map((filter, idx) => {
-          return (
+          return !filter.options ? (
```

Comment:
Rather than having two Button selects that we can return, we can just do some condition logic in the props. Something like options={filter.options.length !== 0  ? emptyFilterOption : filter.options}.

 I also think we should check on the array length of options rather than if options is provided. A querry will normally return an empty array rather than returning nothing so checking the length will prevent the empty state from showing not showing on an empty array.

Reply:
We definitely need to check the length of the array (if it exists). Good catch.

Whether to have two Button selects or use conditions ... I believe the existing approach is more readable, while your suggestion is more concise and not so prone to errors.